core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -333,6 +350,28 @@ private static KeyPair generateRsaKeyPair() {
     return generator.generateKeyPair();
   }
 
+  /** Sets the default string which is appended to the SQLAdmin API client User-Agent header. */
+  public static void addUserAgent(String artifactId) {
+    if (version != null) {
+      userAgents.add(artifactId + "/" + version);
+    } else {
+      userAgents.add(artifactId);
+    }
+  }
+
+  /** Returns the default string which is appended to the SQLAdmin API client User-Agent header. */
+  public static String getUserAgentString() {
+    return String.join(" ", userAgents);
```

Comment:
Should this add in `USER_TOKEN_PROPERTY_NAME` if set?

Reply:
Not sure. Right now I'm inlining combining the two strings when creating the SQLAdmin API client.

Reply:
Never mind, changed this to append the application name in the latest set of commits